### PR TITLE
Coach accessibility improvements

### LINKED ISSUE #9849: Automatic Syncing Updates: Accessibility Remediations in Coach page
## Overview 

This issue is a frontend issue that updates the report and plan tabs in the Coach plugin with improved accessibility features. The pages should: 
* Include the new [TabGroup](https://github.com/learningequality/kolibri-design-system/issues/385) component created in KDS
* Have new visually hidden `h2` tags that will use semantic HTML to group the information architecture of the page that is currently only grouped _visually_

### Figma 

Please see [complete figma spec](https://www.figma.com/file/MpCG7r5PULOCJsEvnTSti8/Android-Designs-2022?node-id=960%3A16641&t=yKUs0H8sJ8FrRN6r-0) for all relevant screens and details. Below is just one example: 
Screen Shot 2022-11-21 at 3 21 07 PM

## Acceptance Criteria
- [ ]  Reports Page has been refactored to include the new KDS TabGroup component
- [ ]  Plan Page has been refactored to include the new KDS TabGroup component
- [ ] When viewing a lesson or quiz, the side panel has a visually-hidden h2: “General information”
- [ ] When viewing a lesson or quiz, the main content has a visually-hidden h2: “Details”
- [ ] THIS **MUST** pass accessibility QA from @radinamatic before it can get merged. _Her approval is blocking._

## References 
- [Notion project page](https://www.notion.so/learningequality/Remote-Content-Syncing-Automatic-syncing-d0a28f150b324d4da019790228117068)
- [Figma spec](https://www.figma.com/file/MpCG7r5PULOCJsEvnTSti8/Android-Designs-2022?node-id=194%3A2922&t=yKUs0H8sJ8FrRN6r-0)
- [Gherkin scenarios](https://www.notion.so/learningequality/Gherkin-scenarios-02096ac9925844f8902c818cab607d4b#45dcdae16db8450b98023b559724277f)

### PR DESCRIPTION
**Pull request by MisRob**
## Summary

Improves accessibility of _Reports_ and _Plan_ pages in _Coach_ by
- Replacing tabs with new KDS components
- Introducing visually hidden titles to sub-sections on lesson and quiz pages

## References

- Closes #9849
- [Figma designs including visually hidden titles](https://www.figma.com/file/MpCG7r5PULOCJsEvnTSti8/Android-Designs-2022?node-id=194-2922&t=rYUskRSq93qeBS8p-0)

## Reviewer guidance

- [x] Test the following tabs (visual and functional regressions, keyboard navigation, markup in regards to a11y)
  - _Lessons-Quizzes-Groups-Learners_ tabs in _Coach_ → _Reports_
  - _Lessons-Quizzes-Groups_ in _Coach_ → _Plan_
  - _Reports-Members-Activity_ in _Coach_ → _Reports_ → _Groups_ → select a group → the group page

- [x] Test the visually hidden h2 titles  _General information_ and _Details_ on _Reports_ lesson pages
  - _Coach_ → _Reports_ → _Lessons_ → select a lesson → the lesson page
  - _Coach_ → _Reports_ → _Groups_ → select a group → in _Lessons assigned_, select a lesson → the lesson page
  - Figma reference
Screenshot from 2023-03-08 16-23-08

- [x] Test the visually hidden h2 titles  _General information_ and _Details_ on _Reports_ quiz pages
  - _Coach_ → _Reports_ → _Quizzes_ → select a quiz → the quiz page
  - _Coach_ → _Reports_ → _Groups_ → select a group → in _Quizzes assigned_, select a quiz → the quiz page
  - Figma reference
Screenshot from 2023-03-08 16-27-45

- [x] Test the visually hidden h2 title  _General information_ on a _Plan_ lesson page
  - _Coach_ → _Plan_ → _Lessons_ → select a lesson → the lesson page
  - Figma reference
Screenshot from 2023-03-08 16-29-18

- [x] Test the visually hidden h2 title  _General information_ on a _Plan_ quiz page
  - _Coach_ → _Plan_ → _Quizzes_ → select a quiz → the quiz page
  - Figma reference
Screenshot from 2023-03-08 16-33-23

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] ~If there are any front-end changes, before/after screenshots are included~
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by MisRob** [comment]
I'm not sure yet why the Licenses check failed, it doesn't seem to be related to code changes, but I will check on it later.

File: kolibri/plugins/coach/assets/src/views/plan/CoachExamsPage/index.vue
Comment (MisRob) [inline kolibri/plugins/coach/assets/src/views/plan/CoachExamsPage/index.vue:15]: If that'd make review easier, I'm fairly confident I didn't make any updates to the code that is now newly wrapped within `KTabsPanel`, and these large diffs are caused by deeper nesting that resulted from that. This applies to all other files containing `KTabsPanel`.

**Comment by rtibbles** [comment]
The license check failure here has been fixed on develop, so this should be fine to merge, irrespective of that.

**Review by akolson — APPROVED** [review]
HI @MisRob! Great job on this, pr generally LGTM!

**Comment by MisRob** [comment]
Thank you, @akolson.

@radinamatic I know we did some testing already as I was developing the component. However, this PR introduces it to more places and also has some logic that wasn't in the previous work. Would it make sense to test it before merging?

**Comment by radinamatic** [comment]
Hey @MisRob, apologies for the delay! 🙏🏽 

@pcenov did the regression testing and everything seems to be running smooth 👍🏽 

Regarding the keyboard navigation and markup in regards to a11y, I tested the visually hidden headings and the new below implemented tabs, all good! 💯 

- Lessons-Quizzes-Groups-Learners tabs in Coach → Reports
- Lessons-Quizzes-Groups in Coach → Plan
- Reports-Members-Activity in Coach → Reports → Groups → select a group → the group page
    
@marcellamaki reminded me that the scope of this PR covered only the above pages, so could I ask you to open a follow up issue to also **implement the tabs on the following pages**:
    
- Report-Difficult questions in Coach → Reports → Quizzes → select a quiz → quiz page    
- Report-Learners in Coach → Reports → Lessons → select a lesson → lesson page
- Reports-Activity in Coach → Reports → Learners → select a learner → learner page

**Review by radinamatic — APPROVED** [review]
More accessible tabs, yey! Thank you @MisRob 👏🏽 💯 :shipit:

**Comment by MisRob** [comment]
Please do not merge yet, I need to test it after resolving conflicts. I will merge after I'm done.

**Comment by MisRob** [comment]
@radinamatic Thank you. Yes, I can open the follow-up issues.

**Comment by MisRob** [comment]
Follow-up issues
- https://github.com/learningequality/kolibri/issues/10253
- https://github.com/learningequality/kolibri/issues/10254
- https://github.com/learningequality/kolibri/issues/10255

**Pull request merged**
